pathfinding_sdf/pathfinding_sdf.c

```diff
@@ -0,0 +1,670 @@
+            float y = p1.y + dy * t + 0.5f;
+            if (points != NULL && pointCount > 0)
+            {
+                Vector2 fan[32];
+                for (int j=0;j<pointCount;j++)
+                {
+                    float px = points[j].x * radius + x * cellSize;
+                    float py = points[j].y * radius + y * cellSize;
+                    fan[j] = (Vector2){ px, py };
+                }
+                DrawTriangleFan(fan, pointCount, color);
+            }
+            else
+                DrawCircle(x * cellSize, y * cellSize, radius, color);
+            
+            return;
+        }
+        pointDistance += d;
+    }
+    *walkedPathDistance = 0.0f;
+} 
+
+float CalcPathLength(PathfindingNode* path, int pathCount)
+{
+    float length = 0.0f;
+    for (int i=1;i<pathCount;i++)
+    {
+        PathfindingNode p1 = path[i-1];
+        PathfindingNode p2 = path[i];
+        float dx = p2.x - p1.x;
+        float dy = p2.y - p1.y;
+        length += sqrtf(dx * dx + dy * dy);
+    }
+    return length;
+}
+
+//------------------------------------------------------------------------------------
+// Program main entry point
+//------------------------------------------------------------------------------------
+int main(void)
+{
+    SetTraceLogLevel(LOG_ALL);
+    // Initialization
+    //--------------------------------------------------------------------------------------
+    const int screenWidth = 800;
+    const int screenHeight = 450;
+
+    InitWindow(screenWidth, screenHeight, "raylib [sdf pathfinding] example");
+
+    SetTargetFPS(60);
+
+    // initialize square root lookup table for cheap square root calculation
+    for (int i=0;i<256;i++)
+    {
+        isqrt[i] = (int)ceilf(sqrtf(i));
+    }
+
+    for (int x = -10; x <= 10; x++)
+    {
+        for (int y = -10; y <= 10; y++)
+        {
+            int d = isqrt[x * x + y * y];
+            if (d <= 10 && d > 0)
+            {
+                neighborOffsets[neighborOffsetCount] = (NeighborOffset){ x, y, d };
+                neighborOffsetCount++;
+            }
+        }
+    }
+
+    const Color gridColor = (Color){ 200, 200, 200, 40 };
+    const Color cellHighlightColor = (Color){ 200, 0, 0, 80 };
+    const float movementSpeed = 3.0f;
+
+    blockedCells = (char *)MemAlloc(gridWidth * gridHeight * sizeof(char));
+    sdfCells = (char *)MemAlloc(gridWidth * gridHeight * sizeof(char));
+
+    // rat setup
+    int pathRatStartX = 5;
+    int pathRatStartY = 25;
+    int pathRatEndX = 75;
+    int pathRatEndY = 25;
+    PathfindingNode *mapRat = (PathfindingNode *)MemAlloc(gridWidth * gridHeight * sizeof(PathfindingNode));
+    PathfindingNode *pathRat = (PathfindingNode *)MemAlloc(gridWidth * gridHeight * sizeof(PathfindingNode) * 4);
+    int pathRatLength = 0;
+    int ratWallFactor = 2;
+
+    // cat setup
+    int pathCatStartX = 5;
+    int pathCatStartY = 25;
+    int pathCatEndX = 75;
+    int pathCatEndY = 25;

```

Comment:
This is a lot of arguments, many that are just X and Y variants.
Can we use vector2 or some other structure to clean this up?

Reply:
True. I can create a dedicated struct for each entity. It would be nice if raylib had a Vector2Int... I wanted to keep it integer based (not sure if worth it).

Reply:
Everything infernally uses floats, so is there any real benefit to keeping it ints? If you need to you can just call floor on the floats.

Reply:
True. It was a decision based on gut feeling. I would keep it for now. I moved the data into structs so I hope it's more organized now.